google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/SetAdParameters.java

```diff
@@ -0,0 +1,137 @@
+ * To get ad group criteria, run GetKeywords.java
+ * */
+public class SetAdParameters {
+
+  private static class SetAdParameterParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private Long adGroupId;
+
+    @Parameter(names = ArgumentNames.CRITERION_ID, required = true)
+    private Long criterionId;
+  }
+
+  public static void main(String[] args) throws IOException {
+    SetAdParameters.SetAdParameterParams params = new SetAdParameters.SetAdParameterParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.adGroupId = Long.parseLong("INSERT_AD_GROUP_ID_HERE");
+      params.criterionId = Long.parseLong("INSERT_CRITERION_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+        "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new SetAdParameters().runExample(googleAdsClient, params.customerId, params.adGroupId, params.criterionId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+        "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+        gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupId the ad group ID.
+   * @param criterionId criterion ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId, long criterionId) {
+    String adGroupCriterionResourceName = ResourceNames.adGroupCriterion(customerId, adGroupId, criterionId);
+
+    // Create ad parameters.
+    // There can be a maximum of two AdParameters per ad group criterion.
```

Comment:
Can you explain what the parameter index is doing here?

Reply:
The AdParameter resource takes an index, but it can only be 1 or 2 because each AdGroupCriterion was designed to only accept up to 2 AdParameters.

Reply:
Sorry my comment may have not been clear - I mean could you add comments to the CL which includes this information to guide the reader